ipaserver/install/installutils.py

```diff
@@ -259,6 +270,35 @@ def record_in_hosts(ip, host_name=None, conf_file=paths.HOSTS):
 
     return None
 
+
+def order_hosts_file(host_name):
+    """Read and overwrite /etc/hosts with FQDN at beginning, issue #8018"""
```

Comment:
This seems to cry out for using ipalib.install.sysrestore, a handle just isn't currently available KRA use of the ReplicaInstall class. This would permanently (perhaps) update the file which we try not to do. It may be straightforward to use sysrestore in the KRA installer, or it may already have a handle somewhere, but this function could be simplified to use sysrestore to handle the backup and then just update the file.

Reply:
Hmmm, so can you explain to me what you are thinking here exactly? you dont like the idea of reorder hosts file by replacing it? This thread is getting extense and probably my use case is not so relevant. Do you think that I should close this issue?

Reply:
IPA tries to be able to restore a system to the previous state. See ipaserver/install/installutils.py::update_hosts_file() for an example of updating the hosts file with a backup (in fact, perhaps this is a better place to do the re-ordering).

Reply:
We appreciate the contribution and realize that sometimes the review process can some onerous.

Reply:
I think we (or I, lol) discuss that here: https://github.com/freeipa/freeipa/pull/3434#issuecomment-516069629. As I said there, perhaps am I wrong, you tell me.

> We appreciate the contribution and realize that sometimes the review process can some onerous.
 I am here to learn and contribute with something, I really appreciate all your feedback and learn a lot from them. :)